backend/inspirehep/records/marshmallow/literature/bibtex.py

```diff
@@ -66,17 +66,19 @@ class BibTexCommonSchema(Schema):
 
     @staticmethod
     def get_date(data, doc_type):
+
         publication_year = BibTexCommonSchema.get_best_publication_info(data).get(
             "year"
         )
         thesis_date = get_value(data, "thesis_info.date")
         imprint_date = get_value(data, "imprints.date[0]")
-
-        if doc_type.endswith("thesis"):
-            date_choice = thesis_date or publication_year or imprint_date
-        else:
-            date_choice = publication_year or thesis_date or imprint_date
-
+        earliest_date = data.earliest_date
+        date_map = {
+            "thesis": thesis_date,
+            "book": imprint_date,
+            "book chapter": imprint_date,
```

Comment:
this is wrong now, as `doc_type` is a bibtex `doc_type`. You should have tests catching this.